### PR TITLE
Single Sign On docs

### LINKED ISSUE #147: Create SSO docs
We need a page to explain all types of SSO supported. (currently required oAuth for Google apps, and SAML with Okta are available).

The SAML configuration page in the webapp, has a "learn more" link that points to `/general-reference/sso#saml`

### LINKED ISSUE #203: Create SSO documentation
Some resources thay may or may not be useful:
- Primary documentation by Isma in [Google Docs](https://docs.google.com/document/d/1jrf7IICs-V7zXxtIrfs0g0wuuq3nDDXLWSSNkXtImeA)
- A really nice [video](https://www.youtube.com/watch?v=XKrnbUFW8Ps) from Bonusly explaining how their integration works 
- RBAC [release notes](https://docs.google.com/document/d/1chWcyTDpMVcmWfvETmeiSBb5VdPLfEAUhkMCwJYJFNs)

### PR DESCRIPTION
**Pull request by iambrosi**
General reference about Authentication and Single Sign On

Closes #203 
Closes #147

### CODE REVIEW

**Comment by mtrias** [comment]
Question: From which stage is the screenshot in `vc_1.png` ?

**Comment by iambrosi** [comment]
@mtrias Prod, all prod.

**Comment by mtrias** [comment]
@Preetam can you please review ?

File: content/general-reference/authentication.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+title: "Authentication"
+maintitle: "Authentication"
+docs: "reference"
+weight: "6"
+---
+
+This page explains the different options you can choose for authenticating when using VividCortex. In order to configure this for your account, you need to have administration privileges.
```
Comment (Preetam) [inline content/general-reference/authentication.md]: `when using` -> `into`?

File: content/general-reference/authentication.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+### Default
+
+The default authentication setting allows users in your team to login using their e-mail and password. If your company uses Google email addresses, they can also authenticate using OAuth with their Google account.
```
Comment (Preetam) [inline content/general-reference/authentication.md]: `login` -> `log in` for consistency

File: content/general-reference/authentication.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+These authentication methods only work for accounts created beforehand, i.e., you need to invite people to your team for them to get access to VividCortex.
+
+### Require OAuth
+
+This configuration allows (and also requires) users in your team to authenticate via Google OAuth using your company's Google apps domain email address.
```
Comment (Preetam) [inline content/general-reference/authentication.md]: Google Apps is now G Suite.

File: content/general-reference/authentication.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+In case of an emergency, such as losing access to your Google account, administrators can still log in using their existing email and password, but everyone else, is required to use OAuth.
```
Comment (Preetam) [inline content/general-reference/authentication.md]: `but everyone else, is required to use OAuth.` -> `but others are required to use OAuth.`

File: content/general-reference/authentication.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+When this configuration is set, users don't need to be created beforehand. When a new person joins your Google apps domain, they can already authenticate to VividCortex, with default privileges.
```
Comment (Preetam) [inline content/general-reference/authentication.md]: Remove comma before `with default privileges`.

File: content/general-reference/authentication.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+### SAML SSO
+
+If you use SAML Single Sign On, all your team is provisioned (and deprovisioned) in real time from a central ID provider of your organization.
```
Comment (Preetam) [inline content/general-reference/authentication.md]: `Single Sign-On`

File: content/general-reference/sso.md [hunk 7]
```diff
@@ -0,0 +1,59 @@
+---
+title: "Single Sign On"
```
Comment (Preetam) [inline content/general-reference/sso.md]: `Sign-On`

Comment (Preetam) [inline content/general-reference/sso.md]: Same comment in other places. I won't mention the others.

File: content/general-reference/sso.md [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+Now go to your Okta Dashboard, find the VividCortex app and add it. A configuration screen will appear.
+
+![Add VividCortex to Okta](/img/docs/sso/okta_1.png)
+
+In the first step, you need to configure the subdomain of your organization in _VividCortex_.
+
+![Setup subdomain](/img/docs/sso/okta_2.png)
+
+If you access _VividCortex_ at https://acme.app.vividcortex.com, then your subdomain is **acme**.
```
Comment (Preetam) [inline content/general-reference/sso.md]: No need for `then`.

File: content/general-reference/sso.md [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+The next step requires you to enable provisioning features. This is mandatory so VividCortex can be notified about any changes in your team.
+
+![Setup credentials](/img/docs/sso/okta_3.png)
+
+Next, you need to enter the _API Authentication token_ for Okta to communicate with VividCortex. You can find this token in the Authentication Settings, back in the VividCortex app.
```
Comment (Preetam) [inline content/general-reference/sso.md]: Remove comma before `back`.

File: content/general-reference/sso.md [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+Make sure the _Okta username format_ is set to **Email address** under User Import. New users created in VividCortex will be downloaded and turned in to new AppUser objects, for matching against existing OKTA users.
```
Comment (Preetam) [inline content/general-reference/sso.md]: `in to` -> `into`

Comment (Preetam) [inline content/general-reference/sso.md]: Remove comma before `for matching against`.

Comment (Preetam) [inline content/general-reference/sso.md]: Why is Okta in caps?

File: content/general-reference/sso.md [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+![User import](/img/docs/sso/okta_4.png)
+
+**Create Users**, **Update User Attributes** and **Deactivate Users** must be enabled in order to maintain the user directory in sync with VividCortex.
+
+![Provisioning features](/img/docs/sso/okta_5.png)
+
+* **Created Users:** New users created through OKTA will also be created in the VividCortex application.
```
Comment (Preetam) [inline content/general-reference/sso.md]: I don't think Okta should be in caps.

File: content/general-reference/sso.md [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+In the final step, you can assign existing users to the VividCortex app. Just check on each user you want to assign access to VividCortex and click *Next* to go to the review step. You can change these assignments at any moment after you finish the configuration.
+
+![User assignment](/img/docs/sso/okta_6.png)
+
+Once you are done, go to the *Sign On* tab and click on *View Setup Instructions*, where you can get the *Federation metadata* required to finish the process in *VividCortex*.
+
+![View Setup Instructions](/img/docs/sso/okta_7.png)
+
+Go back to VividCortex and paste the metadata from the previous step into the text box and click *Save*.
```
Comment (Preetam) [inline content/general-reference/sso.md]: `Go back to VividCortex, paste the metadata from the previous step into the text box, and click *Save*.`

**Comment by iambrosi** [comment]
@Preetam done!

**Comment by mtrias** [comment]
Thanks for the review Preetam!

**Pull request merged**